File: fieldsync-inventory-backend/src/main/java/com/example/fieldsync_inventory_api/entity/sales/CustomerEntity.java
package com.example.fieldsync_inventory_api.entity.sales;

import jakarta.persistence.*;
import lombok.*;
import com.example.fieldsync_inventory_api.entity.Auditable;

import java.time.Instant;

@Entity
@Table(name = "customer")
@Getter
@Setter
@ToString(callSuper = true)
@EqualsAndHashCode(callSuper = true)
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class CustomerEntity extends Auditable {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    // Many-to-one relationship with CustomerType
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "customer_type_id")
    private CustomerTypeEntity customerType;

    @Column(name = "full_name", length = 100, nullable = false)
    private String fullName;

    @Column(length = 50)
    private String email;

    @Column(length = 20, nullable = false)
    private String phone;

    @Column(columnDefinition = "TEXT")
    private String address;

    @Column(name = "deleted_at")
    private Instant deletedAt;
}

File: fieldsync-inventory-app/app/src/main/java/com/example/fieldsync_inventory_app/data/local/entity/CustomerEntity.kt
package com.example.fieldsync_inventory_app.data.local.entity

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.PrimaryKey

@Entity(tableName = "customer")
data class CustomerEntity(
    @PrimaryKey val id: Int,
    @ColumnInfo(name = "customer_type_id")
    val customerTypeId: Int,
    @ColumnInfo(name = "customer_type_name")
    val customerTypeName: String,
    @ColumnInfo(name = "full_name")
    val fullName: String,
    val email: String?,
    val phone: String?,
    val address: String?,
    @ColumnInfo(name = "created_at")
    val createdAt: Long,
    @ColumnInfo(name = "updated_at")
    val updatedAt: Long,
    @ColumnInfo(name = "deleted_at")
    val deletedAt: Long?
)